vortex-array/src/array/compute.rs

```diff
@@ -0,0 +1,12 @@
+use crate::compute::{ComputeFn, KernelRef};
+
+/// A trait used to register static kernels for known compute functions.
+/// Dynamic kernels must be returned via the `_find_kernel` method.
+pub trait ArrayComputeImpl {
+    const FILTER: Option<KernelRef> = None;
```

Comment:
don't sure I understand how the default is supposed to work. I think you can explain a little bit more about the end state and it could make sense but sans commentary there's some weirdness here

Reply:
Basically the `Impl` things are just short-cuts to make implementing the Array trait easier. These consts allow you to return static lifetime kernels by wrapping up an `impl FilterKernel`.

The fallback to `_find_kernel` is in case an array wants to return dynamic kernels (i.e. Arc::new)